web/js/duplicates.js

```diff
@@ -35,6 +40,11 @@
             url_params.pin_size = 'normal';
         }
 
+        if (params && params.category ) {
+            dismissed = params.category === dismissed_category;
+            dismissed_category = params.category;
+        }
```

Comment:
Why not return here if it's going to be dismissed, before making a request to the server?